CHANGELOG.md

```diff
@@ -0,0 +1,72 @@
+# V2.0.0
+
+## New features
+Most new features are centered around better typings
+
+### EventDispatcher can be restricted to certain event classes
+You can now pass a generic parameter to `EventDispatcher` indicating
+which type of events it will dispatch. Besides it being documentation to
```

Comment:
```suggestion
which type of events it will dispatch. Besides it being documented to
```

Reply:
This is meant to say 'documentation' :) But I'll rephrase it as 'Besides it _serving as_ documentation to...' maybe that's more clear.